src/test/java/org/sagebionetworks/web/unitclient/widget/entity/EntityListRowBadgeTest.java

```diff
@@ -24,52 +26,70 @@
 import org.sagebionetworks.repo.model.FileEntity;
 import org.sagebionetworks.repo.model.Project;
 import org.sagebionetworks.repo.model.Reference;
+import org.sagebionetworks.repo.model.file.S3FileHandle;
 import org.sagebionetworks.schema.adapter.JSONObjectAdapterException;
 import org.sagebionetworks.web.client.DateTimeUtils;
+import org.sagebionetworks.web.client.PopupUtilsView;
+import org.sagebionetworks.web.client.SynapseJSNIUtils;
 import org.sagebionetworks.web.client.SynapseJavascriptClient;
+import org.sagebionetworks.web.client.events.DownloadListUpdatedEvent;
 import org.sagebionetworks.web.client.utils.Callback;
+import org.sagebionetworks.web.client.widget.entity.EntityBadge;
 import org.sagebionetworks.web.client.widget.entity.EntityListRowBadge;
 import org.sagebionetworks.web.client.widget.entity.EntityListRowBadgeView;
-import org.sagebionetworks.web.client.widget.entity.file.FileDownloadButton;
+import org.sagebionetworks.web.client.widget.entity.file.AddToDownloadList;
+import org.sagebionetworks.web.client.widget.entity.file.FileDownloadMenuItem;
 import org.sagebionetworks.web.client.widget.lazyload.LazyLoadHelper;
 import org.sagebionetworks.web.client.widget.user.UserBadge;
 import org.sagebionetworks.web.test.helper.AsyncMockStubber;
 
+import com.google.gwt.event.shared.EventBus;
 import com.google.gwt.user.client.rpc.AsyncCallback;
 import com.google.gwt.user.client.ui.Widget;
 
 public class EntityListRowBadgeTest {
 
 	EntityListRowBadge widget;
 	String entityId = "syn123";
+	String entityName = "an entity";
 	@Mock
 	EntityListRowBadgeView mockView;
 	@Mock
 	UserBadge mockUserBadge;
 	@Mock
 	DateTimeUtils mockDateTimeUtils;
 	@Mock
-	FileDownloadButton mockFileDownloadButton;
-	@Mock
 	LazyLoadHelper mockLazyLoadHelper;
 	@Mock
 	SynapseJavascriptClient mockSynapseJavascriptClient;
+	@Mock
+	PopupUtilsView mockPopupUtils;
+	@Mock
+	EventBus mockEventBus;
+	@Mock
+	SynapseJSNIUtils mockSynapseJSNIUtils;
+	@Mock
+	FileEntity mockFileEntity;
+	@Mock
+	S3FileHandle mockDataFileHandle;
 	@Before
 	public void before() throws JSONObjectAdapterException {
```

Comment:
Check entity list widget, make sure add to download list looks (and functions) ok.